state-and-event-handling/event-handling.md

```diff
@@ -0,0 +1,182 @@
+
+```
+Hello! We're in printMessage!
+```
+
+### Event Handling Functions Can Accept `event`
+
+We can get details about our events in our event-handling functions.
+
+If our event-handling function accepts arguments, the first argument will be an event object.
+
+Consider this code, where `printMessage` accepts an event object, and then prints it.
+
+<!-- prettier-ignore-start -->
+```js
+const Post = () => {
+    const printMessage = (event) => {
+        console.log('Hello! We\'re in printMessage!');
+        console.log('This event object contains details about the event:', event);
+    }
+
+    return (<section>
+        <button onClick={printMessage}>Like</button>
+    </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+This is what prints to the browser console. (Your results may vary depending on your browser.)
+
+```
+Hello! We're in printMessage!
+This event object contains details about the event: 
+Object { _reactName: "onClick", ...
+```
+
+### Event Handling Functions Are Functions
+
+It's helpful to know that our event handling functions are, otherwise, normal functions.
+
+They can call other functions inside them, and pass arguments to them, too.
+
+Consider this `Post` component. It:
+
+- Defines a function, `printName`, which takes in a `name` argument
+- Defines another function, `printMessage`. It calls `printName` with the argument `"Ada Lovelace"`.
+- Sets `printMessage` as the click event handler for a "Like" button
+
+<!-- prettier-ignore-start -->
+```js
+const Post = () => {
+    const printName = (name) => {
+        console.log('We\'re in printName!');
+        console.log(`Hello, ${name}!`);
+    }
+
+    const printMessage = () => {
+        printName('Ada Lovelace');
+        console.log('Now, we\'re in printMessage!');
+    }
+
+    return (<section>
+        <button onClick={printMessage}>Like</button>
+    </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+Our output proves that our `printMessage` event handler can invoke other functions.
+
+```
+We're in printName!
+Hello, Ada Lovelace!
+Now, we're in printMessage!
+```
+
+## Pattern: Anonymous Functions
+
+Our event handlers can be _anonymous functions_.
+
+Consider our `printMessage` and `printName` example. Let's refactor our `printMessage` into an inline, anonymous function:
+
+<!-- prettier-ignore-start -->
+```js
+const Post = () => {
+    const printName = (name) => {
+        console.log('We\'re in printName!');
+        console.log(`Hello, ${name}!`);
+    }
+
+    return (<section>
+        <button onClick={() => printName('Ada Lovelace')}>Like</button>
+    </section>)
+}
+```
+<!-- prettier-ignore-end -->
+
+This code produces the output:
+
+```
+We're in printName!
+Hello, Ada Lovelace!
+```
+
+Every time we click on our "like" button, our anonymous function will fire. This, in turn, invokes `printName('Ada Lovelace')`!
+
+This is a common pattern in React code.
+
+When our event handler has at least one parameter, we can use this technique.
```

Comment:
expand this part of the explanation too. recognize this pattern works well in some situations, but named functions still work better in others.
```suggestion
So when our desired behavior takes at least one parameter, we can use this technique of wrapping it in an anonymous function.

An anonymous function lets us call arbitrary code exactly where the event registration occurs, which can be good for understandability. But we may find that, if we need to do more than one or two things in the event handler, then using a named function may still be more understandable.
```